include/assert.h

```diff
@@ -43,8 +44,7 @@ void _assert_fail(
 #endif /* DEBUG */
 
 /* Create an assert that will trigger a compile error if it fails. */
-#define compile_assert(name, expr) \
-        typedef int __assert_failed_##name[(expr) ? 1 : -1] UNUSED;
+#define compile_assert(name, expr) SEL4_COMPILE_ASSERT(name, expr)
```

Comment:
The kernel is only supposed to use type and constant definitions from libsel4 when they relate to communication between the kernel and user level and so it doesn't make sense for the kernel's private definition of compile_assert to be coupled to the implementation in libsel4.

Reply:
Ok, I will remove this change then. But I would like to change this to use `_Static_assert()` (or whatever native way the current compiler implements) whenever this exists. The error messages are just nicer there and it's a bit less hacky. Also, on the pragmatic side both gcc and clang support `_Static_assert()` even when `-std=c99` is used

to use the definition from libsel4 here then and see that whenever the compiler support a native compile time assert, this is preferred. That's a bit more convenient regarding the error messages